packages/block-library/src/query-pagination-next/index.php

```diff
@@ -49,10 +49,17 @@ function render_block_core_query_pagination_next( $attributes, $content, $block
 		$custom_query           = new WP_Query( build_query_vars_from_query_block( $block, $page ) );
 		$custom_query_max_pages = (int) $custom_query->max_num_pages;
 		if ( $custom_query_max_pages && $custom_query_max_pages !== $page ) {
+			$data_attributes = sprintf(
+				'data-block="%1$s" data-attribute="%2$s" data-new-value="%3$s"',
+				'core/query',
```

Comment:
The assumption is that the block that controls the interaction (here: pagination; could also be some other interaction, such as a filter) lives inside of the block that renders the thing that we want to change (here: the list of posts). We're specifying the type of that ancestor block here.